src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
```

Comment:
I haven't followed all of the discussion around this, but I think the div alternative might be the right one at this point. It should solve most cases without too much headache. We should mention in the README that a div is added automatically, but that might be enough for this to become useful.